plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/statistics/DeltaLakeJsonFileStatistics.java

```
@@ -141,7 +142,15 @@ private Optional<Object> deserializeStatisticsValue(DeltaLakeColumnHandle column
         if (!columnHandle.isBaseColumn()) {
             return Optional.empty();
         }
-        Object columnValue = deserializeColumnValue(columnHandle, statValue, DeltaLakeJsonFileStatistics::readStatisticsTimestamp, DeltaLakeJsonFileStatistics::readStatisticsTimestampWithZone);
+
+        Object columnValue;
+        try {
+            columnValue = deserializeColumnValue(columnHandle, statValue, DeltaLakeJsonFileStatistics::readStatisticsTimestamp, DeltaLakeJsonFileStatistics::readStatisticsTimestampWithZone);
```

Is it hard to add support for the varbinary case ? I would rather we do that than add a try/catch which will hide the underlying problem

---

@raunaqmorarka I'm not entirely sure, and adding the stats here has serious correctness implications (skip a partition that shouldn't be skipped -> incomplete data 😬 ), so I don't feel confident I can come up with a good solution and would rather explicitly skip unsupported statistics in this PR, and let someone else come up with the enhancement in a separate PR.

Additionally, besides VARBINARY, any implementation could come up with statistics for new types, so this PR future-proofs Trino to ensure we'll keep using the stats we know how to handle and avoid the yet-unimplemented ones.

---

@ebyhr @findinpath do you know how to tackle adding support for varbinary stats ? I'd rather have that than swallowing exceptions

---

@raunaqmorarka adding support for varbinary would fix this specific issue, but I still think we should handle unknown types more gracefully, as currently a failure means that _all stats are ignored_, rather than just the unknown stat being ignored. Wdyt?

---

Hey folks, happy new year! :) Hope you had a happy holiday season. Trying to get this moved forward, I still strongly feel it's better to have two PRs here, this one to handle unknown statistics gracefully, and a follow-up for varbinary stats; wdyt @raunaqmorarka / @ebyhr / @findinpath ?

---

If we skip the stats, then do we have chance to parse some know formats of the stats, like we know the default format of timestamp in Spark in the future?
So, next question is how to add those `enhancement` in the future, or at least we should file an issue, or add some comments?

---

I've added a code comment and a TODO